### DISCUSSION TITLE
How do I make the animation of the image movement smooth?

### DISCUSSION THREAD

**Discussion by romikforest** [ManimCommunity/manim#3638] [Q&A]
If I add some image and try to move it (or alternatively move the camera) my animation looks pretty smooth with a small run_time. If I set the run_time for example to 30 seconds, the movement looks jerky and strange (borders are twitching etc.). Is there any way in manim to ensure that the picture moves smoothly for long run_time?

Sample code:

```
from manim import *  
  
  
class SampleY(MovingCameraScene):  
  
    def construct(self):    
        image1 = ImageMobject('img1.webp').scale(1.5)  
  
        self.add(image1)  
  
        frame = self.camera.frame  
        self.play(frame.animate(run_time=30).scale(0.8).shift(2 * UR))
```

**Comment by uwezi** [comment]
I tested your code and can confirm that it is a bit rough when rendering in low-quality, but it gets better if you render at high-quality. And the probable reason: this is caused by numerical inaccuracies - you are trying to move the image a very short distance, i.e. not many pixels, but a movement by a fraction of a pixel cannot be rendered correctly. In low-quality the rendered video has only 854x480 pixels resolution. Just looking at the shifting and ignoring the scaling this means that you are moving in total 120 pixels over the time of 30 seconds or 4 pixels per second, about one pixel every fourth rendered frame.
In high-quality the screen resolution is about 2x larger in both axes, the movement is about 8 pixels per second and the pixels are much smaller.
The uneven squishing is at least partly caused by your scaling to 80% also over the stretch of 30 seconds, where Manim cannot generate fractional pixel sizes for the scaled image.

Everything is also emphasized by the smooth animation curve, which starts at zero velocity, accelerates and then decelerates again. So in the beginning there is even less then a 4 pixel per second movement.

**Reply by romikforest** [reply]
How exactly does it work? The original image has a size of 1532x920 pixels. When manim loads it, does it convert to fit the screen size internally? I can't get smooth motion at any resolution of the output file. Even worse, there are vertical stripes in the picture, they, along with the edges of the picture, are clearly deformed during the movement and the picture moves in jerks.

**Comment by uwezi** [comment]
Manim has to scale the pixel matrix of your image to match the pixel matrix in the output frame for the video. It can do so using different scaling algorithms - however, repeated scaling of a pixel-based object will deteriorate the quality with every rescaling... 
See here: https://docs.manim.community/en/stable/reference/manim.mobject.types.image_mobject.ImageMobject.html?highlight=image#imageinterpolationex
